lib/controllers/patients/core.js

```diff
@@ -25,31 +27,31 @@ var initialFilterInput = crud.filterInputGenerator(initialKeys),
     returnListData = crud.returnListData;
 
 // Create new patient with current user given write access by default
-patients.post("/", auth.authenticate, initialFilterInput, function (req, res, next) {
+patients.post("/", auth.authenticate, guard(["admin"]), initialFilterInput, function (req, res, next) {
     Patient.createForUser(req.data, req.user, returnData(res, next));
 }, formatObjectCode(201)); // return 201 status code
 
 // View single patient
-patients.get("/:id", auth.authenticate, function (req, res, next) {
+patients.get("/:id", auth.authenticate, guard(["admin", "programAdministator", "clinician", "user"]), function (req, res, next) {
     // Requires user to have at least read access to patient
     Patient.findByIdForUser(req.params.id, req.user, "read", returnData(res, next));
 }, formatObject);
 
 // Update single patient
-patients.put("/:id", auth.authenticate, filterInput, function (req, res, next) {
+patients.put("/:id", auth.authenticate, guard(["admin", "programAdministator", "clinician", "user"]), filterInput, function (req, res, next) {
     // Requires user to have write access to patient
     Patient.findByIdAndUpdateForUser(req.params.id, req.data, req.user, returnData(res, next));
 }, formatObject);
 
 // Delete single patient
-patients.delete("/:id", auth.authenticate, function (req, res, next) {
+patients.delete("/:id", auth.authenticate, guard(["admin", "programAdministator", "clinician", "user"]), function (req, res, next) {
```

Comment:
should users be able to delete their files?

Reply:
I guess it would be truly difficult for them to do this, and their would be bigger problems with compliance with the program than just them deleting their patient record

Reply:
This is a legal question. I don't know.